floss/plugins/arithmetic_plugin.py

```diff
@@ -62,3 +62,34 @@ def is_security_cookie(self, f, n_bb, n_instr):
 
     def score(self, function_vas, vivisect_workspace=None):
         return function_vas  # scoring simply means identifying functions with non-zero XOR instructions
+
+
+class ShiftPlugin(plugin_object.GeneralPlugin):
+    """
+    Identify shift instructions.
+    """
+    implements = [interfaces.DecodingRoutineIdentifier]
+    version = 1.0
+
+    def identify(self, vivisect_workspace, fvas):
+        candidate_functions = {}
+        for fva in fvas:
+            f = viv_utils.Function(vivisect_workspace, fva)
+            mnems = set([])
+            shift_mnems = set(["shl", "shr", "sar", "sal", "rol", "ror"])
+            for bb in f.basic_blocks:
+                try:
+                    for i in bb.instructions:
+                        mnems.add(i.mnem)
+                        if i.mnem in shift_mnems:
+                            self.d("shift instruction: %s va: 0x%x function: 0x%x", i, i.va, f.va)
+                except envi.InvalidInstruction:
+                    self.w("Invalid instruction encountered in basic block, skipping: 0x%x", bb.va)
+                    continue
+
+            candidate_functions[fva] = 1 - (len(shift_mnems - mnems) / float(len(shift_mnems)))
```

Comment:
So, the more different instructions we find in a function, the more likely it's a decoding routine. What about many of the same shift operation (e.g. a lot of `shl` operationss?)